aries_cloudagent/messaging/jsonld/routes.py

```diff
@@ -6,23 +6,37 @@
 from marshmallow import fields
 
 from ...admin.request_context import AdminRequestContext
-from ...wallet.base import BaseWallet
 
 from ..models.openapi import OpenAPISchema
-
+from ...config.base import InjectionError
+from ...wallet.error import WalletError
+from ...resolver.did_resolver import DIDResolver
+from ...resolver.base import ResolverError
+from .error import MissingVerificationMethodError, BaseJSONLDMessagingError
 from .credential import sign_credential, verify_credential
-from .error import (
-    BadJWSHeaderError,
-    DroppedAttributeError,
-    MissingVerificationMethodError,
-)
 
 
 class SignRequestSchema(OpenAPISchema):
     """Request schema for signing a jsonld doc."""
 
     verkey = fields.Str(required=True, description="verkey to use for signing")
-    doc = fields.Dict(required=True, description="JSON-LD Doc to sign")
+    doc = fields.Dict(
+        required=True,
+        description="JSON-LD Doc to sign",
+        doc=fields.Dict(
+            credential=fields.Dict(
+                required=True,
+                description="credential to sign",
+            ),
+            options=fields.Dict(
+                description="option describing how to sign",
+                required=True,
+                creator=fields.Str(required=False),
+                verificationMethod=fields.Str(required=False),
+                proofPurpose=fields.Str(required=False),
+            ),
+        ),
+    )
```

Comment:
You have to declare a separate schema for this to work; the `fields.Dict` field only allows you to declare what the keys will look like (i.e. str) and what the values will look like (i.e. str). The `credential` and `options` kwargs are getting stored as metadata on the dictionary field and are not used during validation.

Reply:
```python
doc=fields.Nested(Schema.from_dict({
    "credential": fields.Dict(
        required=True,
        description="credential to sign",
    ),
    "options": fields.Nested(Schema.from_dict({
            creator=fields.Str(required=False),
            verificationMethod=fields.Str(required=False),
            proofPurpose=fields.Str(required=False),
        }), 
        description="option describing how to sign",
        required=True,
    ),
})
```

Reply:
I'm not sure that this is advisable but this is similar to what it would need to look like.

Reply:
`Schema.from_dict({` 
ahhh. 
I will try again.